### ISSUE TITLE
offline maps always updated on mobile data connections

### ISSUE THREAD

**Issue by twojstaryzdomu** [osmandapp/OsmAnd#15365]
### Description
Offline maps get pulled on all types of internet connections indiscriminately, which will lead to charges or used internet quota on metered networks for many users. An option to avoid updates on mobile data networks is a key necessity here.

Filing this under bug due to exposure and likely financial impact or monthly data quota loss.

Downloading large offline (several hundred MBs worth of data) on mobile internet defeats the purpose users enable them for. The chief reason is usually to avoid charges or running out of monthly quota.

It can be especially problematic abroad, case in point a user that has specifically set up offline maps to avoid roaming network charges, only to find osmmap auto-download offline maps for updates. The bills for data in some regions while on roaming mobile networks abroad can be huge.

What is more, offline map updates cannot be stopped from the information pop-up on Android 12. So if a user finds their map is updating and wants to stop it, there is no way to stop the process half-way through. To stop updating, the user needs to open the app and find the right menu option, which given offline maps are not under the settings menu, can difficult. And it is a hassle to enable updates back when on wifi.

### How to reproduce?
Set up offline maps updates. Use osmand irregularly to have a map becoming obsolete. Enable data connection over mobile network. Enable osmand to observe offline maps updated.

### Your Environment
OsmAnd Version: 4.1.11
Android/iOS version: 12 and earlier
Device model: all

**Maps used (online or offline):**   
Not an issue specific to a particular map.

**Comment by scaidermern** [comment]
Is this about OsmAnd live? For each live map there is a switch to restrict updates to wifi only.

**Comment by tradavyvy** [comment]
I use osmand+ and my maps only update when I specificly tell osmand to update (wifi or mobile data). And that is precicely how I want it.

**Comment by twojstaryzdomu** [comment]
Well not inherently the live component, only a global download on wifi only option that would prevent unintended updates whilst on mobile data. The per-map download on wifi seems to be hidden all the way through, behind several option screens. Users will struggle to find it unless without guidance such an option exists.

Can a global download on wifi only toggle be brought forward to make it more user-friendly? Many users will likely run into the same scenario.

Also, it doesn't seem possible to be able to cancel an ongoing download in any way, other than hard stopping the app It might be another issue to fix.

**Comment by twojstaryzdomu** [comment]
> Is this about OsmAnd live? For each live map there is a switch to restrict updates to wifi only.

Well not inherently the live component, only a global download on wifi only option that would prevent unintended updates whilst on mobile data. The per-map download on wifi seems to be hidden all the way through, behind several option screens. Users will struggle to find it unless without guidance such an option exists.

Can a global download on wifi only toggle be brought forward to make it more user-friendly? Many users will likely run into the same scenario.

Also, it doesn't seem possible to be able to cancel an ongoing download in any way, other than hard stopping the app's background data usage. It might be another issue to fix.

**Comment by sonora** [comment]
It seems a bit unclear what your issue is: Maps only update when you explicitly start a download (or update) on the Download screen. And each ongoing download has an (X) button there, where you can cancel it any time?

![1](https://user-images.githubusercontent.com/845735/192111446-33e3061e-0e2b-40f6-aaf1-4d346f656e5d.jpg)

And you can always get to the download screen when you tap the download notification on the Android system notification:

![Untitled](https://user-images.githubusercontent.com/845735/192111479-9e7efa2e-af73-4468-9208-d3d7aa3269b9.jpg)

**Comment by twojstaryzdomu** [comment]
> And each ongoing download has an (X) button there, where you can cancel it any time?
I was hoping to find it in the android pop-up. Some pop-ups do have buttons. Don't recall seeing it in the live update section.

**Comment by vshcherb** [comment]
There is a setting to update or not update OsmAnd Live using Mobile connection

**Comment by vshcherb** [comment]
I'm afraind I misunderstood the issue if it's about warning that you are about to download on mobile data, I think it's a duplicate issue

**Comment by twojstaryzdomu** [comment]
Of which is this a duplicate? My concern is for there not being a global option to disable all mobile downloads quickly. With many maps, it is a chore to have to toggle them back and forth as needed.

**Comment by vshcherb** [comment]
Yes there is no such thing unfortunately, I believe it should be implemented on Android level

**Comment by twojstaryzdomu** [comment]
> Why do you need to change this setting so often? Either you want map updates on mobile data or you don't want them.

Simple. Border crossing case, no desire to pay for large map downloads while abroad WITHOUT having to:
1. Toggle updates several maps on and off every time I am on roaming data network abroad.
2. Disable data altogether merely to prevent maps from being updated.

People can cross borders often so the setting matter. I'm sure you can think of other use cases for a global switch.

Discretionary per-map update switches are overkill. Also, most other apps that allow updates of a component feature a single global mobile data off switch. It doesn't need to be as fine-grained.

**Comment by sonora** [comment]
You can prevebt apps from background-using data in (Android-)Settings/Apps/Osmand/Allow background data usage.

**Comment by sonora** [comment]
This issue initially sounded like there is some bug with all offline maps updating automatically and uncontrollably, but from what I see now there is nothing left here that is not already in #13856.

**Comment by twojstaryzdomu** [comment]
> I see now there is nothing left here that is not already in #13856.

Where? I don't see anything related to a global mobile data usage killswitch for updates. Besides, that bug  a 'nice to have'. I can reword this bug specifically for the sake of users who travel across their national borders and are often subject to roaming data charges but would like it opened back, not dismessed under a catch-all request relating to data.

**Comment by mwerle** [comment]
Just found this thread while searching something similar as well.

I DO NOT want OsmAnd to automatically update maps or anything else ever. Live maps or otherwise. All updates or synchronisations should be explicit.

Sometimes I briefly connect my navigation device to the internet and if it then downloads a partial map set, map versions end up out of sync and it can't route anymore.

Of course I understand other people do want automated updates and sync. So there should be an app-wide setting to disable or enable automated downloads/updates. 

If this is already an option, I can't find it, please point it out.

Thank you for considering.

**Comment by sonora** [comment]
@mwerle To my knowledge:
- The recent issue that you cannot route anymore with mixed map versions has been fixed.
- OsmAnd does never auto-update the standard offline maps unless you explicitly trigger an update manually
- For OsmAnd Live there is a toggle on top of the Maps & Resources 'Updates' tab.

**Comment by scaidermern** [comment]
> * For OsmAnd Live there is a toggle on top of the Maps & Resources 'Updates' tab.

I think there is a design flaw for live update settings. Each map has its own live update settings for enabling live updates, configuring the update interval and deciding whether updates on wifi should be allowed. Apart from the obvious problem that many maps lead to many settings that can be / have to be changed individually there is also another questionable decision: Live updates via wifi are disabled by default. This leads to updates being downloaded immediately via mobile connection as soon as the user enables live updates for a certain map and no wifi connection is present.

**Comment by sonora** [comment]
Ah, in that case it's probably worth a separate and dedicated issue... at the bottom of this old and closed one it is likely to get lost... :wink:

**Comment by mwerle** [comment]
Sorry, I had actually started #25148 after making the comment in this closed thread.

Honestly I wasn't expecting anybody to be notified on this one anymore ;)  As I pointed out in that other issue, I think there are genuine usability concerns in finding and changing the live update settings. My immediate issue is resolved, but I think there is room for improvement. I can close that issue and open a new one to concretely discuss the usability if you like, or we can use it to discuss further.

In either case, yes agree, further discussion on an already-closed issue is not ideal; my apologies - I didn't actually realise it was already closed until I went to hit "Comment" and at that point I didn't feel like losing everything I had typed already..
